src/aurelia-event-aggregator.ts

```diff
@@ -91,7 +91,7 @@ export class EventAggregator {
    * @param event The event channel or event data type.
    * @param callback The callback to be invoked when when the specified message is published.
    */
-  public subscribe(event: string | Constructor, callback: EventCallback): Subscription {
+  public subscribe<T = any>(event: string | Constructor, callback: EventCallback<T>): Subscription {
```

Comment:
What is the added value of this? I still have to specify the `data` type for the callback.
If we just look at the *constructor as topic* overload, this signature seems better `public subscribe<T>(event: Constructor<T>, callback: EventCallback<T>): Subscription;`. Now just by passing the constructor I get the `data` type inferred. You should be able to do something similar for the *string as topic* overload and the optional parameter.
Why not explore an overloads approach? Later on it can be merged into single signature again, if it isn't too bloated.

Reply:
Well the added value was for `string` type events. For messages I forgot to add the generic. An oversight which will be fixed shortly. Thank you for pointing it out.